src/UI/Component/Button/Engageable.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+/* Copyright (c) 2018 Nils Haagen <nils.haagen@concepts-and-training.de> Extended GPL, see docs/LICENSE */
+
+namespace ILIAS\UI\Component\Button;
+
+/**
+ * A Button can be stateful; when engaged, the state MUST be refelected
+ * in the "aria-pressed"-, respectively the "aria-checked"-attribute.
+ * If the Button is not stateful (which is the default), the aria-attribute can
+ * be omitted.
+ * https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/button_role
```

Comment:
This is a rule that MUST be added to the rules of the parent button. Since engaged is not available on all buttons, you may rephrase this to something like: "Some Buttons can be stateful; when ...."

Reply:
Also, some phrase in the effect part of buttons should hint to this as well. Lastly add an example showing the use of this engaged state on standard buttons.